Linting/fix rviz rendering followup

anhosi opened the pull request:
Addresses follow up issues of #34 that were raised in pr #67.

The renaming of the headers to hpp is required for a up-to-date cppcheck which we introduced in the Windows build.

anhosi committed:
Remove NOLINT comments

- replace using namespace statements

anhosi committed:
Improve expressiveness of NOLINT comment

anhosi committed:
Remove ./ from header includes

- As a NOLINT is necessary for these includes anyway there is no need
  for the ./ as path (introduced in pr #67)
- The NOLINT can be removed once the corresponding headers are renamed
  to a .hpp.

anhosi committed:
Restore include order

- disable cpplint where necessary

anhosi committed:
Rename all headers to hpp file ending

A current cppcheck (windows) parses .h files as c files instead of c++ files

anhosi committed:
Adapt code to moving of header files

anhosi commented:
There is a remaining code style issue: relative paths in includes, sometimes even in the the form `./header.hpp`. As the google style guide https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes also suggests to not use this notation I would like to replace that. What do you think @wjwwood ?

wjwwood commented on rviz_common/src/rviz_common/frame_manager.hpp:
```diff
@@ -31,18 +31,18 @@
-#include <mutex>
-#include <string>
+#include <geometry_msgs/msg/pose.hpp> // NOLINT: cpplint is unable to handle the include order here
+#include <rclcpp/time.hpp> // NOLINT: cpplint is unable to handle the include order here
```
This shouldn't pass uncrustify (or maybe it's cpplint and that's why it doesn't show up), but you need two spaces between code and comment.

Also you can just use `"` here instead of `<` to shut cpplint up.

wjwwood replied:
Same in other files.

By the way I didn't do this in the Qt headers case because that goes against all of the Qt examples so I was constantly changing the include style and there was a lot of existing instances. But for our libraries we're including, this style is probably better.

A harder, but more awesome 😄, solution would be to extend our tool `ament_cpplint` (and it's embedded copy of cpplint.py) to not be so picky in these cases.

wjwwood replied:
Please check the two spaces before trailing comments in other places where `NOLINT` is used too.

anhosi replied:
For now I will go with `"` for ros includes.

wjwwood commented on rviz_common/src/rviz_common/temp/stl_loader.cpp:
```diff
@@ -80,13 +80,13 @@ bool STLLoader::load(const std::string & path)
   // find the file size
   fseek(input, 0, SEEK_END);
-  long fileSize = ftell(input);  // NOLINT maybe use int64 instead
+  long fileSize = ftell(input); // NOLINT ftell returns long
```
Please do not remove the extra space (!) 😢, that is the cpplint style and the preferred way according to the ROS 2 style. Here and elsewhere in the pr.

anhosi committed:
Switch to " style includes for ros headers

- This avoids cpplint include order errors
- Also fixing few include order issues

anhosi committed:
Restore double spaces for line comments

- In case of NOLINT comments the linter does not pick up on the wrong style.
